api/src/test/java/com/epam/pipeline/acl/cluster/ClusterApiServiceTest.java

```diff
@@ -0,0 +1,451 @@
+    private final FilterNodesVO filterNodesVO = NodeCreatorUtils.getDefaultFilterNodesVO();
+    private final NodeDisk nodeDisk = NodeCreatorUtils.getDefaultNodeDisk();
+    private final MonitoringStats monitoringStats = ClusterCreatorUtils.getMonitoringStats();
+    private final InstanceType instanceType = NodeCreatorUtils.getDefaultInstanceType();
+    private final InputStream inputStream = new ByteArrayInputStream(TEST_STRING.getBytes());
+    private final Authentication authentication = new TestingAuthenticationToken(new Object(), new Object());
+
+    private final List<NodeDisk> nodeDisks = Collections.singletonList(nodeDisk);
+    private final List<InstanceType> instanceTypes = Collections.singletonList(instanceType);
+    private final List<MonitoringStats> statsList = Collections.singletonList(monitoringStats);
+
+    @Autowired
+    private ClusterApiService clusterApiService;
+
+    @Autowired
+    private NodesManager mockNodesManager;
+
+    @Autowired
+    private PipelineRunManager mockPipelineRunManager;
+
+    @Autowired
+    private AuthManager mockAuthManager;
+
+    @Autowired
+    private UsageMonitoringManager mockUsageMonitoringManager;
+
+    @Autowired
+    private NodeDiskManager mockNodeDiskManager;
+
+    @Autowired
+    private InstanceOfferManager mockInstanceOfferManager;
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnListWithNodeInstancesForAdmin() {
+        doReturn(initSingleNodeInstanceList()).when(mockNodesManager).getNodes();
+
+        assertThat(clusterApiService.getNodes()).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnListWithNodeInstancesWhenPermissionIsGranted() {
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(initSingleNodeInstanceList()).when(mockNodesManager).getNodes();
+
+        assertThat(clusterApiService.getNodes()).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnListWithNodeInstanceWhichPermissionIsGranted() {
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        doReturn(authentication).when(mockAuthManager).getAuthentication();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        initAclEntity(nodeInstanceWithoutPermission, AclPermission.NO_READ);
+        doReturn(initTwoNodeInstancesList()).when(mockNodesManager).getNodes();
+
+        assertThat(clusterApiService.getNodes()).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnEmptyNodeInstanceListWhenPermissionIsNotGranted() {
+        initAclEntity(nodeInstanceWithoutPermission);
+        doReturn(initSingleNodeInstanceList()).when(mockNodesManager).getNodes();
+
+        assertThat(clusterApiService.getNodes()).isEmpty();
```

Comment:
Method `initSingleNodeInstanceList()` returns list which doesn't contain `nodeInstanceWithoutPermission` so this test is not testing what it should. Probably we can even get rid of methods like `initSingleNodeInstanceList` and always initialiaze lists like the following
```java
Collections.singletonList(node);
// or
Arrays.asList(node1, node2);
```